jerry-core/ecma/operations/ecma-regexp-object.c

```diff
@@ -1140,6 +1247,9 @@ re_match_regexp (re_matcher_ctx_t *re_ctx_p, /**< RegExp matcher context */
           lit_utf8_read_prev (&str_curr_p);
           num_of_iter--;
         }
+#ifdef REGEXP_RECURSION_LIMIT
+        --re_ctx_p->recursion_depth;
```

Comment:
There are lots of --re_ctx_p->recursion_depth. Would it be less operations if it is called after a recursive call?

Reply:
You are right, it would be less operations. I will update the PR with this change.